### PR TITLE
DKI Updates: (new radial tensor kurtosis metric, updated documentation and missing tests)

### PR DESCRIPTION
**Pull request by RafaelNH**
Hi all. 

I added a new DKI metric - the radial tensor kurtosis proposed by [Hansen et al., 2017](https://pubmed.ncbi.nlm.nih.gov/27539807/).

**Why another DKI metric?**
This radial tensor kurtosis metric is more robust to implausible negative kurtosis estimates than the standard radial kurtosis definition - see my simulation results in my repository [Radial_Tensor_Kurtosis](https://github.com/RafaelNH/Radial_Tensor_Kurtosis/blob/main/Radial_tensor_kurtosis_comparison_in_simulations.ipynb).

**Update in DOC/TESTS** I also used this opportunity to update some function documentation and tutorials of DKI (e.g. add reference to the paper reporting our DKI implementations). I also added tests for some lines of code that were not previously covered by tests - we have 100% test coverage for dki.py again!

Please let me know your comments/suggestions.

### CODE REVIEW

**Review by skoudoro — COMMENTED** [review]
Thank you for this @RafaelNH,

See below a quick review. I need to go a bit deeper.

Also, on the tutorial, Can you add a note to explain why user can have "holes" on DKI result.  This is a question that we get often in general and we also got it multiple times during the workshop.

Also, Could you have a look at this dki question: https://github.com/dipy/dipy/issues/3145?

Thank you

File: dipy/reconst/tests/test_dki.py [hunk 1]
```diff
@@ -209,13 +214,26 @@ def test_dki_fits():
     assert_array_almost_equal(dkiF_multi.model_params, masked_multi_params)
 
     # testing return of S0
-    dki_S0M = dki.DiffusionKurtosisModel(gtab_2s, fit_method="WLS",
-                                         return_S0_hat=True)
-    dki_S0F = dki_S0M.fit(signal_cross)
-    dki_S0F_S0 = dki_S0F.model_S0
+    if have_cvxpy:
+        tested_methods = ['NLS', 'WLS', 'CLS']
```
Comment (samcoveney) [inline dipy/reconst/tests/test_dki.py:218]: Suggest to change NLS to NLLS

**Comment by samcoveney** [comment]
Gone through this, LGTM, thanks @RafaelNH for adding tests for model_S0 for DKI, I forgot to add those in #3078

**Comment by RafaelNH** [comment]
@skoudoro , @samcoveney ! I've addressed all your comments. Let me know if there is something that I can do here.

File: doc/examples/reconst_dki.py [hunk 2]
```diff
@@ -54,6 +65,9 @@
                     & W_{xxzz} & W_{yyzz} & W_{xxyz} & W_{xyyz} & W_{xyzz}
                     & & )\\end{matrix}
 
+Example Tutotial
```
Comment (skoudoro) [inline doc/examples/reconst_dki.py:68]: Typo, but you can replace by `In Practice` or something similar. (avoid "example, tutorial)

File: doc/examples/reconst_dki.py [hunk 3]
```diff
@@ -211,20 +225,49 @@
...
+# signals in DIPY are described in the following tutorial
+# (:ref:`sphx_glr_examples_built_reconstruction_reconst_msdki.py`).
+# 3) computing alternative definitions of mean and radial kurtosis such as
+# the mean kurtosis tensor (MKT) and radial tensor kurtosis (RTK) metrics (see
+# below).
+# 4) constrained optimization to ensure that the fitted parameters
+# are physically plausible [DelaHa2020]_ (see below).
+#
+# Alternative DKI metrics
+# =======================================================
```
Comment (skoudoro) [inline doc/examples/reconst_dki.py:246]: too long, should be the same size as the title

File: doc/examples/reconst_dki.py [hunk 4]
```diff
@@ -211,20 +225,49 @@
...
+# artifacts [Hansen2017]_.
+# 3) the kurtosis fractional anisotropy (KFA) that quantifies the anisotropy of
+# the kurtosis tensor [GlennR2015]_, which provides different information than
+# the FA measures from the diffusion tensor.
+# These measures are computed and illustrated below:
+
+compare_maps([dkifit], ['mkt', 'rtk', 'kfa'], fit_labels=['DKI'],
+             map_kwargs=[{'vmin': 0, 'vmax': 1.5}, {'vmin': 0, 'vmax': 1.5},
+                         {'vmin': 0, 'vmax': 1}],
+             filename='Alternative DKI metrics.png')
```
Comment (skoudoro) [inline doc/examples/reconst_dki.py:264]: Please, can you avoid space for filename. it can be messy  depending on the operating system. Use underscore

**Comment by RafaelNH** [comment]
PS: I replaced the data smoothing step with mppca denoising since I noticed that users where using Gaussian Smoothing for DKI reconstruction of their data (rather than our denoising strategies) just because we were using smoothing in our example.

**Comment by skoudoro** [comment]
is there any additional comments concerning this PR @arokem ? @samcoveney ?  someone else ?

**Comment by samcoveney** [comment]
Happy to take a look over the next 24 hours, but if the tests are okay now, should be okay (so long as you coded the new metric correctly!)

**Comment by RafaelNH** [comment]
> Happy to take a look over the next 24 hours, but if the tests are okay now, should be okay (so long as you coded the new metric correctly!)

Well it is passing the tests for both multi and single voxels (i.e. their estimates matching the ground truth in case we now the GT). Also there was an group that already used this code and compared with their own implementations and got consistent results. Hope that metric is well implemented because that comparison work is already published;)

**Review by skoudoro — APPROVED** [review]
Thank you for this work @RafaelNH.

I am going ahead and merge this PR.

Just a quick note: MPPCA is quite slow and this modification slowdown a lot the tutorial.
I have created an issue to not forget this : https://github.com/dipy/dipy/issues/3193.
Currently, when you run the tutorial, you do not know if your laptop is stuck or if something is really running. 
this is a feeling that need to be fixed.

**Comment by jhlegarreta** [comment]
@RafaelNH thanks for doing this.

Looks like this introduced new warnings:
https://github.com/dipy/dipy/actions/runs/8727767577/job/23946008747#step:10:4402

Any chance to address them, please?

**Pull request merged**